### ISSUE TITLE
Influence 2D projection (seed)

### ISSUE THREAD

**Issue by datrinh** [lmcinnes/umap#242]
Hi,

my problem case is summed up here: https://datascience.stackexchange.com/questions/52391/how-to-reduce-position-changes-after-dimensionality-reduction

Since no1 could help me so far, I'm reaching out in this repo. Is there any way to seed the positions of the 2D projection? If not, how would I go about modifiying the algorithm to my needs?

Similar issues here:
[1] https://stackoverflow.com/questions/30066268/provide-starting-positions-to-t-distributed-stochastic-neighbor-embedding-tsne?rq=1
[2] https://ieeexplore.ieee.org/document/7539329

**Comment by lmcinnes** [comment]
Yes, in fact you can provide the ``init`` argument with either ``'random'``, ``'spectral'`` *or a numpy array of initial positions*. Thus if you have positions for a 2D projection in mind stored an array ``previous_embedding`` then you can do:
```python
new_embedding = umap.UMAP(init=previous_embedding).fit_transform(data)
```
and UMAP will start with the points in the psoitions specified by previous embedding and then optimize from there. Hopefully this is what you are looking for.

**Comment by datrinh** [comment]
@lmcinnes Wow, this sounds just like what I was looking for. I will try it out. Thanks for the awesome lib!

**Comment by datrinh** [comment]
So I tried it out and I'm not sure if I did it right. Images of 4 iterations: https://imgur.com/a/VcDKBVe
How can I check if the init parameter is even working?

**Comment by datrinh** [comment]
@lmcinnes Another thing I have noticed: When I use the init parameter, but use unsupervised dimensionality reduction, it still changes the positions each time. By my intuition the positions shouldn't change in a unsupervised scenario?

**Comment by lmcinnes** [comment]
Ultimately the algorithm runs a stochastic optimization to find positions for points -- that means that, no matter what, there is some randomness, so things will move even if you initialise with the end result of a previous run. This is simply the nature of the fact that the problem to be solved (finding the global minimum) is not tractable, and we are simply taking a path as best we can. Restarting can potentially jump things so we are moving amongst a bunch of local minima. If you want to decrease the amount of movement then vary the ``learning_rate`` parameter to something lower and things will move less.

**Comment by datrinh** [comment]
Thanks for your help. I have realized that recalculating the projection on each step can't work out the way I envisioned it. Im closing this.
